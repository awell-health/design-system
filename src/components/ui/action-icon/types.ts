export enum ActionType {
  ApiCall = 'API_CALL',
  ApiCallGraphql = 'API_CALL_GRAPHQL',
  Calculation = 'CALCULATION',
  Checklist = 'CHECKLIST',
  ClinicalNote = 'CLINICAL_NOTE',
  Form = 'FORM',
  Message = 'MESSAGE',
  Plugin = 'PLUGIN',
  PushToEmr = 'PUSH_TO_EMR'
}
