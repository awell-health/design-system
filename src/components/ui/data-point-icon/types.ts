export type DataPointValueType =
  | 'attachment'
  | 'attachments_array'
  | 'boolean'
  | 'date'
  | 'json'
  | 'number'
  | 'numbers_array'
  | 'string'
  | 'strings_array'
  | 'telephone';
