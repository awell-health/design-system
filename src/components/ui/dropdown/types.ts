export type DropdownItem = {
  label: string | JSX.Element;
  onClick: () => void;
};
