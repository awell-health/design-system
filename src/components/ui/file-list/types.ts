export interface FileListItem {
  name: string;
  id?: string;
  size?: number;
  type?: string;
  progress?: number;
  error?: string;
}
