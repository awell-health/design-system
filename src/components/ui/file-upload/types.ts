export type AcceptedFileTypes = '.jpg' | '.jpeg' | '.png' | '.pdf' | '.doc' | '.docx' | '.txt';
