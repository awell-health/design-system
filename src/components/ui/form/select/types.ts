import { OptionProps, MultiValue, SingleValue } from 'react-select';

export type SelectItem = {
  value: string;
  label: string;
};

export type GroupedOption = {
  label: string;
  options: SelectItem[];
};

export type SelectValue = SingleValue<SelectItem> | MultiValue<SelectItem>;
export type { OptionProps };
