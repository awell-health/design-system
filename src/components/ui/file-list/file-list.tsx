import { FC } from 'react';
import { Button } from '../button/button';
import { Icon } from '../icon';
import { FileListItem } from './types';

interface Props {
  files: Array<FileListItem>;
  onDelete: (file: FileListItem) => void;
  generalProgress?: number;
}

const FileList: FC<Props> = ({ files, onDelete, generalProgress = undefined }) => {
  const renderFileItem = (file: FileListItem, index: number) => {
    const showProgress = !file.error && file.progress && file.progress > 0 && file.progress < 100;
    return (
      <div
        key={index}
        className='flex flex-row items-center gap-4 px-4 py-2 border border-gray-200 rounded-md file-list-item'
      >
        <div>
          <Icon icon='RiFileLine' />
        </div>
        <div className='flex flex-1 flex-col'>
          <div>{file.name}</div>
          <div className='w-full pr-4'>
            {file.error && <div className='text-red-600 text-sm'>{file.error}</div>}
            {showProgress && (
              <progress className='progress progress-info' value={file.progress} max='100' />
            )}
            {!showProgress && !file.error && (
              <div className='text-sm text-gray-500'>
                {file.size ? (file.size / 1024 / 1024).toFixed(2) : '0.00'}MB
              </div>
            )}
          </div>
        </div>
        <div className='ml-auto'>
          <Button variant='secondary' size={null} shape='squareSm' onClick={() => onDelete(file)}>
            <Icon icon='RiDeleteBinLine' size={16} className='fill-red-600' />
          </Button>
        </div>
      </div>
    );
  };
  return (
    <div className='flex flex-col gap-2'>
      {generalProgress && (
        <div className='flex flex-col gap-1 pb-2'>
          <div>Uploading files...</div>
          <progress className='progress progress-info' value={generalProgress} max='100' />
        </div>
      )}
      {files.map((file, index) => renderFileItem(file, index))}
    </div>
  );
};

export { FileList };
