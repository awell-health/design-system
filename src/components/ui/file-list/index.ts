export { FileList } from './file-list';
export type { FileListItem } from './types';
