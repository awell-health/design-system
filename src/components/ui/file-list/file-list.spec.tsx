import React from 'react';
import { describe, expect, it, vi } from 'vitest';
import { fireEvent, render } from '@testing-library/react';
import { FileList } from './file-list';
import { FileListItem } from './types';

const defaultFiles = [
  { name: 'file1.txt', size: 100, type: 'text/plain' },
  { name: 'file2.txt', size: 200, type: 'text/plain' },
  { name: 'file3.txt', size: 300, type: 'text/plain' }
];

describe('FileList Component', () => {
  const subject = (files: FileListItem[], onDelete?: (file: FileListItem) => void) => {
    return <FileList files={files} onDelete={onDelete ?? (() => {})} />;
  };

  it('should mathch snapshot', () => {
    const { container } = render(subject(defaultFiles));
    expect(container).toMatchSnapshot();
  });

  it('should display the correct number of files', () => {
    const { container } = render(subject(defaultFiles));
    expect(container.querySelectorAll('.file-list-item')).toHaveLength(defaultFiles.length);
  });

  it('triggers onDelete when the delete button is clicked', () => {
    const onDelete = vi.fn();
    const { container } = render(subject(defaultFiles, onDelete));
    const deleteButton = container.querySelector('.file-list-item button');

    if (deleteButton) {
      fireEvent.click(deleteButton);
      expect(onDelete).toHaveBeenCalled();
    } else {
      throw new Error('Delete button not found');
    }
  });

  // Add more tests as needed for additional functionality
});
