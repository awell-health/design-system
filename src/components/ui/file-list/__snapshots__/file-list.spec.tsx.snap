// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FileList Component > should mathch snapshot 1`] = `
<div>
  <div
    class="flex flex-col gap-2"
  >
    <div
      class="flex flex-row items-center gap-4 px-4 py-2 border border-gray-200 rounded-md file-list-item"
    >
      <div>
        <svg
          class="remixicon inline"
          fill="default"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 2.00318V2H19.9978C20.5513 2 21 2.45531 21 2.9918V21.0082C21 21.556 20.5551 22 20.0066 22H3.9934C3.44476 22 3 21.5501 3 20.9932V8L9 2.00318ZM5.82918 8H9V4.83086L5.82918 8ZM11 4V9C11 9.55228 10.5523 10 10 10H5V20H19V4H11Z"
          />
        </svg>
      </div>
      <div
        class="flex flex-1 flex-col"
      >
        <div>
          file1.txt
        </div>
        <div
          class="w-full pr-4"
        >
          <div
            class="text-sm text-gray-500"
          >
            0.00
            MB
          </div>
        </div>
      </div>
      <div
        class="ml-auto"
      >
        <button
          class="btn btn-square w-8 h-8 min-w-8 min-h-8 rounded-md btn-secondary bg-white border-slate-300 hover:bg-slate-100 active:bg-slate-100 disabled:text-slate-200 disabled:border-slate-200 disabled:bg-white"
        >
          <svg
            class="remixicon fill-red-600"
            fill="default"
            height="16"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 6H22V8H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V8H2V6H7V3C7 2.44772 7.44772 2 8 2H16C16.5523 2 17 2.44772 17 3V6ZM18 8H6V20H18V8ZM9 11H11V17H9V11ZM13 11H15V17H13V11ZM9 4V6H15V4H9Z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="flex flex-row items-center gap-4 px-4 py-2 border border-gray-200 rounded-md file-list-item"
    >
      <div>
        <svg
          class="remixicon inline"
          fill="default"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 2.00318V2H19.9978C20.5513 2 21 2.45531 21 2.9918V21.0082C21 21.556 20.5551 22 20.0066 22H3.9934C3.44476 22 3 21.5501 3 20.9932V8L9 2.00318ZM5.82918 8H9V4.83086L5.82918 8ZM11 4V9C11 9.55228 10.5523 10 10 10H5V20H19V4H11Z"
          />
        </svg>
      </div>
      <div
        class="flex flex-1 flex-col"
      >
        <div>
          file2.txt
        </div>
        <div
          class="w-full pr-4"
        >
          <div
            class="text-sm text-gray-500"
          >
            0.00
            MB
          </div>
        </div>
      </div>
      <div
        class="ml-auto"
      >
        <button
          class="btn btn-square w-8 h-8 min-w-8 min-h-8 rounded-md btn-secondary bg-white border-slate-300 hover:bg-slate-100 active:bg-slate-100 disabled:text-slate-200 disabled:border-slate-200 disabled:bg-white"
        >
          <svg
            class="remixicon fill-red-600"
            fill="default"
            height="16"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 6H22V8H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V8H2V6H7V3C7 2.44772 7.44772 2 8 2H16C16.5523 2 17 2.44772 17 3V6ZM18 8H6V20H18V8ZM9 11H11V17H9V11ZM13 11H15V17H13V11ZM9 4V6H15V4H9Z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="flex flex-row items-center gap-4 px-4 py-2 border border-gray-200 rounded-md file-list-item"
    >
      <div>
        <svg
          class="remixicon inline"
          fill="default"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 2.00318V2H19.9978C20.5513 2 21 2.45531 21 2.9918V21.0082C21 21.556 20.5551 22 20.0066 22H3.9934C3.44476 22 3 21.5501 3 20.9932V8L9 2.00318ZM5.82918 8H9V4.83086L5.82918 8ZM11 4V9C11 9.55228 10.5523 10 10 10H5V20H19V4H11Z"
          />
        </svg>
      </div>
      <div
        class="flex flex-1 flex-col"
      >
        <div>
          file3.txt
        </div>
        <div
          class="w-full pr-4"
        >
          <div
            class="text-sm text-gray-500"
          >
            0.00
            MB
          </div>
        </div>
      </div>
      <div
        class="ml-auto"
      >
        <button
          class="btn btn-square w-8 h-8 min-w-8 min-h-8 rounded-md btn-secondary bg-white border-slate-300 hover:bg-slate-100 active:bg-slate-100 disabled:text-slate-200 disabled:border-slate-200 disabled:bg-white"
        >
          <svg
            class="remixicon fill-red-600"
            fill="default"
            height="16"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 6H22V8H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V8H2V6H7V3C7 2.44772 7.44772 2 8 2H16C16.5523 2 17 2.44772 17 3V6ZM18 8H6V20H18V8ZM9 11H11V17H9V11ZM13 11H15V17H13V11ZM9 4V6H15V4H9Z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
